reference/configuration/framework.rst

```diff
@@ -193,44 +276,402 @@ see :doc:`/cookbook/request/load_balancer_reverse_proxy`.
+
+When this is set to ``true``, the profiler will only be enabled when an
+exception is thrown during the handling of the request.
+
+only_master_requests
+....................
+
+**type**: ``boolean`` **default**: ``false``
+
+When this is set to ``true``, the profiler will only be enabled on the master
+requests (and not on the subrequests).
+
+dsn
+...
+
+**type**: ``string`` **default**: ``'file:%kernel.cache_dir%/profiler'``
+
+The DSN where to store the profiling information.
+
+.. seealso::
+
+    See :doc:`/cookbook/profiler/storage` for more information about the
+    profiler storage.
+
+username
+........
+
+**type**: ``string`` **default**: ``''``
+
+When needed, the username for the profiling storage.
+
+password
+........
+
+**type**: ``string`` **default**: ``''``
+
+When needed, the password for the profiling storage.
+
+lifetime
+........
+
+**type**: ``integer`` **default**: ``86400``
+
+The lifetime of the profiling storage in seconds. The data will be deleted when
+the lifetime is expired.
+
+matcher
+.......
+
+Matcher options are configured to dynamically enable the profiler. For
+instance, based on the `ip`_ or :ref:`path <reference-profiler-matcher-path>`.
+
+.. seealso::
+
+    See :doc:`/cookbook/profiler/matchers` for more information about using
+    matchers to enable/disable the profiler.
+
+ip
+""
+
+**type**: ``string``
+
+If set, the profiler will only be enabled when the current IP address matches.
+
+.. _reference-profiler-matcher-path:
+
+path
+""""
+
+**type**: ``string``
+
+If set, the profiler will only be enabled when the current path matches.
+
+service
+"""""""
+
+**type**: ``string``
+
+This setting contains the service id of a custom matcher.
+
+router
+~~~~~~
+
+resource
+........
+
+**type**: ``string`` **required**
+
+The path the main routing resource (e.g. a YAML file) that contains the routes
+and imports the router should load.
+
+type
+....
+
+**type**: ``string``
+
+The type of the resource to hint the loaders about the format. This isn't
+needed when you use the default routers with the expected file extensions
+(``.xml``, ``.yml`` / ``.yaml``, ``.php``).
+
+http_port
+.........
+
+**type**: ``integer`` **default**: ``80``
+
+The port for normal http requests (this is used when matching the scheme).
+
+https_port
+..........
+
+**type**: ``integer`` **default**: ``443``
+
+The port for https requests (this is used when matching the scheme).
+
+strict_requirements
+...................
+
+**type**: ``mixed`` **default**: ``true``
+
+Determines the routing generator behaviour. When generating a route that has
+specific :ref:`requirements <book-routing-requirements>`, the generator can
+behave differently in case the used parameters do not meet these requirements.
+
+The value can be one of:
+
+``true``
+    Throw an exception when the requirements are not met;
+``false``
+    Disable exceptions when the requirements are not met and return ``null``
+    instead;
```

Comment:
Yes, that's exactly what this means. I don't know why someone would want to set this either.